src/libraries/Microsoft.PowerFx.Core/Public/Values/VoidValue.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Text;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Types
+{
+    public class VoidValue : ValidFormulaValue
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VoidValue"/> class.
+        /// The void value should be used when the result of an expression is not used or is null.
+        /// </summary>
+        internal VoidValue(IRContext irContext) 
```

Comment:
Would we ever want to be able to recover the IRContext from a Void return value? Should this class be a singleton?

Reply:
This class needs to extend `ValidFormulaValue` class, it's singleton version can be accessed via `FormulaValue.NewVoid()`